community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseBuilder.java

```diff
@@ -39,8 +39,13 @@
  */
 public class GraphDatabaseBuilder
 {
+    @Deprecated
     public interface DatabaseCreator
     {
+        /**
+         * This method will no longer accept a map in 4.0, but a Config object instead.
+         */
+        @Deprecated
         GraphDatabaseService newDatabase( Map<String, String> config );
```

Comment:
You could already now add the `newDatabase( Config )` method, and turn this old declaration into a `default` method implemented as `newDatabase( Config.empty().augment( config ) );`. Or so I think.